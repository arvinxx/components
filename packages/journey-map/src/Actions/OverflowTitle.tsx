import type { FC } from 'react';
import React, { useRef } from 'react';
import { Tooltip } from 'antd';
import { useOverflow } from 'use-overflow';

interface OverflowTitleProps {
  title: string;
}
const OverflowTitle: FC<OverflowTitleProps> = ({ title }) => {
  const textRef = useRef(null);
  const { refYOverflowing } = useOverflow(textRef);
  return (
    <div
      ref={textRef}
      style={{
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
      }}
    >
      {refYOverflowing ? (
        <Tooltip title={title}>{title}</Tooltip>
      ) : (
        <>{title}</>
      )}
    </div>
  );
};

export default OverflowTitle;
