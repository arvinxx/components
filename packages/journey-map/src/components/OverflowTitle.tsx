import type { FC } from 'react';
import React, { useRef } from 'react';
import { Tooltip } from 'antd';
import { useOverflow } from 'use-overflow';

interface OverflowTitleProps {
  title: string;
  direction: 'x' | 'y';
}
const OverflowTitle: FC<OverflowTitleProps> = ({ title, direction = 'x' }) => {
  const textRef = useRef(null);
  const { refYOverflowing, refXOverflowing } = useOverflow(textRef);
  const judged = direction === 'x' ? refXOverflowing : refYOverflowing;

  return (
    <div
      ref={textRef}
      style={{
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
      }}
    >
      {judged ? <Tooltip title={title}>{title}</Tooltip> : <>{title}</>}
    </div>
  );
};

export default OverflowTitle;
