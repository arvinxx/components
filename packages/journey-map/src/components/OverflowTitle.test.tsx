import React from 'react';
import { render } from '@testing-library/react';

import OverflowTitle from './OverflowTitle';

describe('OverflowTitle', () => {
  it('x 轴', () => {
    const { container } = render(<OverflowTitle direction="x" title="hello" />);
    expect(container).toMatchSnapshot();
  });
  it('x 轴 overflow', () => {
    const { container } = render(
      <div style={{ width: 1 }}>
        <OverflowTitle direction="x" title="hello" />
      </div>,
    );
    expect(container).toMatchSnapshot();
  });

  it('y 轴', () => {
    const { container } = render(<OverflowTitle direction="y" title="hello" />);
    expect(container).toMatchSnapshot();
  });
});
