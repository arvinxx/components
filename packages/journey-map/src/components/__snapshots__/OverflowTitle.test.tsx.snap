// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverflowTitle x 轴 1`] = `
<div>
  <div
    style="overflow: hidden; white-space: nowrap; text-overflow: ellipsis;"
  >
    hello
  </div>
</div>
`;

exports[`OverflowTitle x 轴 overflow 1`] = `
<div>
  <div
    style="width: 1px;"
  >
    <div
      style="overflow: hidden; white-space: nowrap; text-overflow: ellipsis;"
    >
      hello
    </div>
  </div>
</div>
`;

exports[`OverflowTitle y 轴 1`] = `
<div>
  <div
    style="overflow: hidden; white-space: nowrap; text-overflow: ellipsis;"
  >
    hello
  </div>
</div>
`;
